client/allocdir/alloc_dir.go

```diff
@@ -174,7 +175,7 @@ func (d *AllocDir) Build(tasks []*structs.Task) error {
 // their intended, relative location within the task directory. Embed attempts
 // hardlink and then defaults to copying. If the path exists on the host and
 // can't be embeded an error is returned.
-func (d *AllocDir) Embed(task string, entries map[string]string) error {
+func (d *AllocDir) Embed(task string, entries map[string]string, roots map[string]string) error {
```

Comment:
Could you create an `embedImpl` method that takes the roots and keep the API of embed the same. `Embed` and then call `embedImpl(entries, entries)`